components/pages/submission-system/program-submitted-data/index.tsx

```diff
@@ -81,64 +80,29 @@ export default function ProgramSubmittedData({ donorId = '' }: { donorId: string
       deSerialize: (v) => v,
     },
   );
+  const currentEntity: string = reverseLookUpEntityAlias(selectedClinicalEntityTab);
   const [selectedDonors, setSelectedDonors] = useUrlParamState('donorId', donorId, {
     serialize: (v) => v,
     deSerialize: (v) => v,
   });
-  const currentEntity: string = reverseLookUpEntityAlias(selectedClinicalEntityTab);
 
-  // Side Menu Query
-  const { data: sideMenuQuery, loading: sideMenuLoading } =
-    FEATURE_SUBMITTED_DATA_ENABLED &&
-    useQuery<ClinicalEntityQueryResponse>(SUBMITTED_DATA_SIDE_MENU_QUERY, {
-      errorPolicy: 'all',
-      variables: {
-        programShortName,
-        filters: defaultClinicalEntityFilters,
-      },
-    });
-
-  useEffect(() => {
-    setGlobalLoading(sideMenuLoading);
-  }, [sideMenuLoading]);
+  const currentDonor = selectedDonors ? [parseDonorIdString(selectedDonors)] : [];
 
-  const { clinicalData: sideMenuData } =
-    sideMenuQuery == undefined || sideMenuLoading ? emptyResponse : sideMenuQuery;
-  const menuItems = clinicalEntityFields.map((entity) => (
-    <VerticalTabs.Item
-      key={entity}
-      active={selectedClinicalEntityTab === aliasedEntityNames[entity]}
-      onClick={(e) => setSelectedClinicalEntityTab(aliasedEntityNames[entity])}
-      disabled={
-        !sideMenuData.clinicalEntities.some((e) => e.entityName === aliasedEntityNames[entity])
-      }
-    >
-      {clinicalEntityDisplayNames[entity]}
-      {hasClinicalErrors(sideMenuData, entity) && (
-        <VerticalTabs.Tag variant="ERROR">!</VerticalTabs.Tag>
-      )}
-    </VerticalTabs.Item>
-  ));
-
-  // Matches sequential Digits not followed by other chars, or Digits preceded by DO or by Comma/Space
-  // Example: DO259138, 2579137, DASH-7 will match first 2 Donor IDs, but not 3rd Submitter ID
-  const searchDonorIds = selectedDonors
-    ? [parseDonorIdString(selectedDonors)]
-    : keyword
-        .match(/(\d*(?!\d*\D+))|((?<=,|, )|(?<=DO))\d*/gi)
-        ?.filter((match) => !!match)
-        .map((idString) => parseInt(idString)) || [];
+  // Matches multiple digits and/or digits preceded by DO, followed by a comma, space, or end of string
+  // Example: DO259138, 2579137, DASH-7, DO253290abcdef
+  // Regex will match first 2 Donor IDs, but not 3rd Submitter ID or 4th case w/ random text
+  const searchDonorIds =
+    keyword
+      .match(/(?=DO|\d)\d+(?=,| |$)/gi)
+      ?.filter((match) => !!match)
+      .map((idString) => parseInt(idString)) || [];
```

Comment:
```suggestion
  const searchDonorIds =
    keyword
      .match(/(?=DO|\d)\d+(?=,| |$)/gi)
      ?.filter((match) => !!match)
      .map((idString) => parseInt(idString)) || [];
```

this could potentially be iterating on undefined as well but I'm not sure where to put the `|| []` without testing it.

Reply:
awesome! learned something new today